src/sources/amqp.rs

```diff
@@ -0,0 +1,565 @@
+                                    let mut stream = stream.map(|event| event.with_batch_notifier(&batch));
+
+                                    match out.send_event_stream(&mut stream).await {
+                                        Err(error) => {
+                                            emit!(StreamClosedError { error, count: 1 });
+                                        }
+                                        Ok(_) => {
+                                            // TODO Is this needed?
+                                            // drop(stream);
+                                            finalizer.add(msg.into(), receiver);
+                                        }
+                                    }
+                                }
+                                None => {
+                                    match out.send_event_stream(&mut stream).await {
+                                        Err(error) => {
+                                            emit!(StreamClosedError { error, count: 1 });
+                                        }
+                                        Ok(_) => {
+                                            let ack_options = lapin::options::BasicAckOptions::default();
+                                            if let Err(error) = msg.acker.ack(ack_options).await {
+                                                error!(message = "Unable to ack.", error = ?error, internal_log_rate_secs = 10);
+                                            }
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                } else {
+                    break
+                }
+            }
+        };
+    }
+
+    Ok(())
+}
+
+async fn handle_ack(status: BatchStatus, entry: FinalizerEntry) {
+    match status {
+        BatchStatus::Delivered => {
+            let ack_options = lapin::options::BasicAckOptions::default();
+            if let Err(error) = entry.acker.ack(ack_options).await {
+                error!(message = "Unable to ack.", error = ?error, internal_log_rate_secs = 10);
+            }
+        }
+        BatchStatus::Errored => {
+            let ack_options = lapin::options::BasicRejectOptions::default();
+            if let Err(error) = entry.acker.reject(ack_options).await {
+                error!(message = "Unable to reject.", error = ?error, internal_log_rate_secs = 10);
+            }
+        }
+        BatchStatus::Rejected => {
+            let ack_options = lapin::options::BasicRejectOptions::default();
+            if let Err(error) = entry.acker.reject(ack_options).await {
+                error!(message = "Unable to reject.", error = ?error, internal_log_rate_secs = 10);
+            }
+        }
+    }
+}
+
+#[cfg(test)]
+pub mod test {
```

Comment:
It looks like neither the mod test nor integration_test are utilizing the `assert_source_compliance` or `assert_source_error` test helpers.